starknet_py/net/client_models.py

```diff
@@ -332,7 +335,25 @@ class ContractClass:
 
     program: dict
     entry_points_by_type: EntryPointsByType
-    abi: Optional[AbiDictList] = None
+    abi: Optional[AbiDictList]
+
+
+@dataclass
+class CompiledContract(ContractClass):
+    """
+    Dataclass representing contract compiled through starknet-compile.
+    """
+
+    abi: AbiDictList
+
+    @classmethod
+    def from_contract(cls, contract: str) -> CompiledContract:
+        """
+        Creates CompiledContract from already compiled contract.
+
+        :return: a CompiledContract instance.
+        """
+        return cast(CompiledContract, CompiledContractSchema().loads(contract))
```

Comment:
Actually, if we're not using `cls` at all here, this could be changed to `@staticmethod`.

Or we could write 
```suggestion
    @classmethod
    def from_contract(cls, contract: str) -> CompiledContract:
        """
        Creates CompiledContract from already compiled contract.

        :return: a CompiledContract instance.
        """
        return cast(cls, CompiledContractSchema().loads(contract))
```